### DISCUSSION TITLE
Problems consuming a StreamProvider inside a dropDownCallback function

### DISCUSSION THREAD

**Discussion by YaserHip** [rrousselGit/riverpod#2285] [Q&A]
Good day to all, hope you can help me with this, im recently working on some test projects with flutter, riverpod and appwrite. 
Right now I need some help consuming a StremProvider when i select an item from a dropDownButton.
The problem here is that the provider doesnt send the updates when i consume it inside the dropDownCallBack, for references i have something like this:

**This is the callback from the dropdown where i want to listen the stream from the stream provider, and the value printed is the one that doesnt do anything:**

void dropDownCallBack(){
          final stream = ref.watch(
            stringStreamProvider("someStringWithAnID").stream);
          stream.listen((value) {
          print("value: ${value.toString()}");
        });
}

**Here is the StreamProvider, inside the map function i print the event, and that does appear on the console while testing it**

final stringStreamProvider =
    StreamProvider.autoDispose.family<String, String>((ref, list) {

  final repository = ref.watch(repositoryProvider);

  return repository.getDataStream(list).stream.map((event){
       print('insideMap: ${event.payload.toString()}');
       return event.payload.toString();
   })
});

This is the problem, hope someone can help me, i think i have tried everything i know 😞

**Comment by rrousselGit** [comment]
Rather than ref.watch(provider.stream).listen, you probably should do ref.listen(provider)

**Comment by YaserHip** [comment]
Thank you for your quick response 😄.

I have this structure:
```
class Page extends ConsumerStatefulWidget {
  const Page({super.key});

  @override
  ConsumerState<ConsumerStatefulWidget> createState() => Page();
}

class _PageMapState extends ConsumerState<PageMap> {

   void dropDownCallBack(){
      ref.listen(stringStreamProvider("someStringWithAnID"), (previous, next) { print("value: ${next.value}");})
   }

  @override
  Widget build(BuildContext context) {
      return CustomDropDown(
            width: MediaQuery.of(context).size.width,
            margin: const EdgeInsets.fromLTRB(10, 12, 70, 5),
            changed: dropDownCallBack,
            value: _dropDownValue,
            items: prodList
                .map<DropdownMenuItem<String>>((e) => DropdownMenuItem(
                    value: e.id,
                    child: Text(
                     e.name,
                ))) .toList(),
                );
   }
}
```

But i got this error 😞 
ref.listen can only be used within the build method of a ConsumerWidget

**Reply by rrousselGit** [reply]
Sorry I meant to say "ref.listenManual"

**Reply by YaserHip** [reply]
thank you so much!

**Comment by YaserHip** [comment]
Sorry for bothering you again @rrousselGit 😞, I have just implemented the listenManual, and it worked 😆, but i have another problem now, the thing is that cant do another requests with the same provider, it saids that the subscription is null over and over again:

As you can see here, everytime i select an item from the dropdownbutton i do a request from the repository, but this only work the first time, after that the subscription only saids null:
```
void dropDownCallBack(){
      ref.listenManual(stringStreamProvider("someStringWithAnID"), (previous, next) { print("value: ${next.value}");})
   }
```


This is a successful subscription, the first time
```
I/flutter (23800): subscription: ws://adasdas.com/v1/realtime?project=634cdbb2f3ef2733fff7&channels%5B%5D=databases.634ce56d957c7ce2a28c.collections.634ce58859290906caed.documents.63b50fea522aae0be729&channels%5B%5D=databases.634ce56d957c7ce2a28c.collections.634ce58859290906caed.documents.63efe89cb0b3efbd7699
```


And this is the second time I select an item, also the first request stopped to listen
```
I/flutter (23800): subscription: null
```

thank you in advance you have helped me a lot!
